apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex

```diff
@@ -465,4 +477,24 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
       _ -> nil
     end
   end
+
+  defp maybe_local_capture_func(analysis, position) do
+    with {:ok, %Zipper{node: {:/, _, [_, {:__block__, _, _}]}} = zipper} <-
+           Ast.zipper_at(analysis.document, position),
+         %Zipper{node: {:&, _, _}} <- Zipper.up(zipper) do
+      {:/, _, [{local_func_name, meta, _}, {:__block__, _, [arity]}]} = zipper.node
+      start_column = meta[:column]
+      function_name_length = local_func_name |> to_string() |> String.length()
+
+      begin_pos = {position.line, start_column}
+      end_pos = {position.line, start_column + function_name_length}
+      range = to_range(analysis.document, begin_pos, end_pos)
```

Comment:
I think `Lexical.Ast.Range.fetch!` would help here
```suggestion
      function_name_length = local_func_name |> to_string() |> String.length()
      range = Ast.Range.fetch!(zipper.node, document)
      range = put_in(range.end.character, range.start.character + function_name_length)
```

I'm also pretty sure if you use zipper.node, you won't need to tack the function name length on the end.
this might just be
```elixir
range = Ast.Range.fetch!(zipper.node)